numpy/lib/arraysetops.py

```diff
@@ -332,14 +338,49 @@ def intersect1d(ar1, ar2, assume_unique=False):
     >>> from functools import reduce
     >>> reduce(np.intersect1d, ([1, 3, 4, 3], [3, 1, 2, 1], [6, 3, 4, 2]))
     array([3])
+    
+    To return the indices of the values common to the input arrays
+    along with the intersected values:
+    >>> x = np.array([1, 1, 2, 3, 4])
+    >>> y = np.array([2, 1, 4, 6])
+    >>> xy, x_ind, y_ind = np.intersect1d(x, y, return_indices=True)
+    >>> x_ind, y_ind
+    (array([0, 2, 4]), array([1, 0, 2]))
+    >>> xy, x[x_ind], y[y_ind]
+    (array([1, 2, 4]), array([1, 2, 4]), array([1, 2, 4]))
+    >>> np.intersect1d([1, 2, 2, 3, 4, 3, 2],[1, 8, 4, 2, 2, 3, 2, 3], return_indices=True)
+    (array([1, 2, 3, 4]), array([0, 1, 3, 4]), array([0, 3, 5, 2]))
```

Comment:
I don't think this example is particularly useful or well explained - I'd be inclined to remove it

Reply:
will do.